std/http/file_server.ts

```diff
@@ -301,9 +301,11 @@ function html(strings: TemplateStringsArray, ...values: unknown[]): string {
 listenAndServe(
   addr,
   async (req): Promise<void> => {
-    const normalizedUrl = posix.normalize(req.url);
-    const decodedUrl = decodeURIComponent(normalizedUrl);
-    const fsPath = posix.join(target, decodedUrl);
+    let normalizedUrl = posix.normalize(req.url);
+    try {
+      normalizedUrl = decodeURIComponent(normalizedUrl);
+    } catch {}
```

Comment:
Actually I'm a bit worried about this unconditional try-catch. Can this be narrowed? Are you sure you want to allow URLs like `http://localhost:4500/http/testdata/%` ? This seems not ideal.

Reply:
We had to do this, in fact I found this bug when I opened deno_website2/public/. There is a resource path with “%”.  I referenced https://github.com/ljharb/qs/blob/master/lib/utils.js#L116

Reply:
Where is the resource with the filename "%" ?

Reply:
https://github.com/denoland/deno_website2/blob/master/public/index.html#L5

Reply:
I'm confused `"%PUBLIC_URL%/favicon.ico"` is replaced, I believe, when the deno_website2 is built.

Also what does this have to do with file_server?

If I try to open that file by itself in chrome I get all sorts of errors:
<img width="963" alt="Screen Shot 2020-02-10 at 12 54 57 PM" src="https://user-images.githubusercontent.com/80/74182312-d2ea9c80-4c10-11ea-9019-1f36d031e8b7.png">

Reply:
If I open the deno_wibesite2/public/ via file_server. It will redirect to the index.html. After that, %PUBLIC_URL% will be requested automatically and at this time file_server will be broken.

Reply:
I see - that makes sense I can repeat the problem:
```
> file_server .
HTTP server listening on http://0.0.0.0:4500/
[2020-02-10 19:41:36] "GET / HTTP/1.1" 200
error: Uncaught URIError: URI malformed
► https://deno.land/std/http/file_server.ts:305:24

305     const decodedUrl = decodeURIComponent(normalizedUrl);
                           ^

    at https://deno.land/std/http/file_server.ts:305:24
    at listenAndServe (https://deno.land/std/http/server.ts:542:5)
```

So can you make the try-catch still throw if the error is not URIError?